app/routes.py

```diff
@@ -108,105 +192,54 @@ def verify():
-
-        # get total number of Has for this ppe
-        try:
-            has = Has.query.filter_by(ppe_id=item.id)
-            for ha in has:
-                total_count_has += ha.count
-        except:
-            total_count_has = 0
-
-        items.append({
-            "sku": item.sku,
-            "desc": item.desc,
-            "img": item.img.decode(),
-            "count": count,
-            "grey_out": count_other > 0,
-            "other": total_count_has
-        })
-
-    hospital = {
-        "hospital_name": user_hospital.name
-    }
-    return render_template('item_base.html', title='Wants', hospital=hospital, state="Wants", items=items)
+    return redirect(url_for('index'))
 
 @app.route('/has', methods=['GET', 'POST'])
 def has():
+    return redirect(url_for('index'))
+
+
+@app.route('/update-ppe/<name>', methods=['POST'])
+def update_ppe(name):
     if not current_user.is_authenticated:
-        return redirect(url_for('login',next='/has'))
+        return redirect(url_for("login", next="/"))
     
-    user = User.query.filter_by(username=current_user.username).first()
-    if not user.is_verified:
-        return render_template("404.html")
+    Model = None
+
+    if name == "wants":
+        Model = Wants
+    elif name == "has":
+        Model = Has
+    else:
+        return 404
+
+    OtherModel = Has if Model is Wants else Wants
     
     user = User.query.filter_by(username=current_user.username).first()
     user_hospital = Hospital.query.filter_by(id=user.hospital_id).first()
-    
-    skus = PPE.query.all()
-    items = []
-    for item in skus:
-        count = 0
-        count_other = 0
-        total_count_wants = 0
-        try:
-            count = Has.query.filter_by(hospital_id=user_hospital.id, ppe_id=item.id).first().count
-        except:
-            count = 0
-       # adding get count of Has
-        try:
-            count_other = Wants.query.filter_by(hospital_id=user_hospital.id, ppe_id=item.id).first().count
-        except:
-            count_other = 0
 
-        # get total number of Wants for this ppe
-        try:
-            wants = Wants.query.filter_by(ppe_id=item.id)
-            for want in wants:
-                total_count_wants += want.count
-        except:
-            total_count_wants = 0
+    for key, quantity in request.form.items():
+        # key format is in ppe-XX for quantities, so make sure we're looking at the right thing.
+        if not key[:4] == "ppe-": continue
+        ppe_id = key[4:]
+        quantity = int(quantity)
 
-        print(str(count) +", "+str(count_other))
-        
-        items.append({
-            "sku": item.sku,
-            "desc": item.desc,
-            "img": item.img.decode(),
-            "count": count,
-            "grey_out": count_other > 0,
-            "other": total_count_wants
-        })
+        current = Model.query.filter_by(hospital_id=user_hospital.id, ppe_id=ppe_id).first()
+
+        if quantity != 0:
```

Comment:
I'm 80% confident this logic prevents the database from getting into an illegal state (ie. there being both a `Has` and a `Wants` for the same hospital/PPE pair).